### PR TITLE
Wait for emulation to stop when destroying

### PR DESCRIPTION
**Pull request by jmezzacappa**
In some environments, the "emulator-stopped" event did not always fire when calling the `destroy()` method. This waits for emulation to finish stopping before continuing with the rest of the destructor.

My use case, for more context:
I'm running v86 in a React app, and the page does not reload between destroying one emulator instance and starting another. Setting event listeners for the "emulator-stopped" event seemed to prevent garbage collection, so it quickly consumed lots of memory and would crash the browser tab.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ function v86(bus, wasm)
     this.running = false;
 
     /** @type {boolean} */
-    this.stopped = false;
+    this.stopping = false;
```
Comment (jmezzacappa) [inline src/main.js:13]: Renaming this was not necessary, so I put it into a separate commit (with explanation) so it would be easier to revert, if desired.

Comment (copy) [reply src/main.js:13]: `stopping` is a better name, thanks.

File: src/browser/starter.js [hunk 2]
```diff
@@ -698,16 +698,31 @@ V86Starter.prototype.run = async function()
...
+        if (!this.cpu_is_running) return resolve();
+        const listener = () => {
+            this.remove_listener("emulator-stopped", listener);
+            resolve();
+        };
+        this.add_listener("emulator-stopped", listener);
+        this.bus.send("cpu-stop");
+    })
+
+    await promise;
```
Comment (copy) [inline src/browser/starter.js:716]:
```suggestion
    if(!this.cpu_is_running)
    {
        return;
    }

    const stopped = new Promise(resolve => {
        const listener = () => {
            this.remove_listener("emulator-stopped", listener);
            resolve();
        };
        this.add_listener("emulator-stopped", listener);
    });

    this.bus.send("cpu-stop");

    await stopped;
```

(I suggest removing the comment, since it expresses the same idea as the code itself. For reference as to why the code is written this way, one can always refer to `git blame`).

**Review by copy — COMMENTED** [review]
This is a good change, thanks. Just one small stylistic suggestion and should be good to merge.

File: src/browser/starter.js (hunk 2 above)
Comment (jmezzacappa) [reply src/browser/starter.js:716]: Ah, right, that is much nicer :)

Would you mind if I move `this.bus.send("cpu-stop");` to line 712, so it's clearly guaranteed to be called immediately after `this.add_listener("emulator-stopped", listener);` within the same anonymous function?

In other words, if I'm debugging some complex race condition, there would be no reason to question whether side effects triggered by `"cpu-stop"` have taken effect before the anonymous function passed into `new Promise()` has been called.

(There's also a good chance that I'm overthinking this, as I have been experimenting with alternative JS runtimes lately 😅)

Comment (jmezzacappa) [reply src/browser/starter.js]: To save time, I went ahead and pushed a commit that applies your suggested changes, and a separate commit that applies my aforementioned modification (moving `this.bus.send("cpu-stop");`). The last one can be reverted if you prefer to stick with the suggested changes :)

File: src/browser/starter.js [hunk 3]
```diff
@@ -698,16 +698,29 @@ V86Starter.prototype.run = async function()
...
+    const stopped = new Promise((resolve) => {
+        const listener = () => {
+            this.remove_listener("emulator-stopped", listener);
+            resolve();
+        };
+        this.add_listener("emulator-stopped", listener);
+        this.bus.send("cpu-stop");
+    });
+
+    await stopped;
```
Comment (copy) [inline src/browser/starter.js:714]: Change to `await new Promise`.

File: src/browser/starter.js [hunk 4]
```diff
@@ -698,16 +698,29 @@ V86Starter.prototype.run = async function()
  */
 V86Starter.prototype.stop = async function()
 {
-    this.bus.send("cpu-stop");
+    if (!this.cpu_is_running) {
+        return;
+    }
+
+    const stopped = new Promise((resolve) => {
```
Comment (copy) [inline src/browser/starter.js:705]: Remove the parenthesis around `(resolve)`

File: src/browser/starter.js [hunk 5]
```diff
@@ -698,16 +698,29 @@ V86Starter.prototype.run = async function()
  */
 V86Starter.prototype.stop = async function()
 {
-    this.bus.send("cpu-stop");
+    if (!this.cpu_is_running) {
+        return;
+    }
```
Comment (copy) [inline src/browser/starter.js:703]: Preserve the style of the other code (remove space after `if`, put `{` on its own line).

**Review by copy — COMMENTED** [review]
Few more stylistic fixes.

File: src/browser/starter.js (hunk 2 above)
Comment (copy) [reply src/browser/starter.js]: It's probably easier to reason about, indeed.

**Comment by copy** [comment]
Squashed and merged, cheers!
